modern/src/Actions.ts

```diff
@@ -29,14 +34,29 @@ export function gotSkinBlob(blob: Blob, store: ModernStore) {
 
 function gotSkinZip(zip: JSZip, store: ModernStore) {
   return async dispatch => {
+    if (!store.getState().sharedMakiTree) {
+      const resp = await fetch(freeform);
+      const blob = await resp.blob();
+      const sharedZip = await JSZip.loadAsync(blob);
+      const sharedXmlTree = await Utils.inlineIncludes(
+        await Utils.readXml(sharedZip, "freeform/xml/wasabi/wasabi.xml"),
+        sharedZip
+      );
+
+      const sharedMakiTree = await initialize(sharedZip, sharedXmlTree);
```

Comment:
It is definitely a bit weird to turn this into a Maki tree, but all our functions are written for maki trees so it needs to be one. Will change with redux state tree approach